torch/distributed/_composable/replicate.py

```diff
@@ -85,8 +86,21 @@ def init_helper(self) -> None:
         self._collect_params(self.module)
         # Only saved for testing
         replicate.state(self.module)._names = self._names
+        if "device_ids" in self.kwargs:
```

Comment:
Should we even preserve the argument to be called `device_ids`? That is a relic from single process multiple device days right?

Can we just call it `device_id` and share the same type as for FSDP?

Reply:
Sure, but device_id won't really mean the same thing in replicate and FSDP then. FSDP does more things with device_id such as move the module to that device if its on CPU. replicate() might do such things in the future, but not at the moment. At the current moment, DDP device_id arg is mostly used only for moving inputs to the right device.

Reply:
I think that difference between `replicate` and `fully_shard` is fine. The frontend arg name being the same might be more important than unifying semantics right now. What do you think?

I was mainly thinking that we do not have to preserve the legacy `device_ids` (plural) if we are defining a new API now.

Reply:
Sure, will switch this to device_id